packages/tooltip/src/Tooltip.tsx

```diff
@@ -257,6 +263,16 @@ function Tooltip({
     existingId,
   ]);
 
+  // If consumer is using Icon or Glyph component as trigger, the tooltip will not be visible as these components do not render their children
+  if (
+    (trigger && isComponentType(trigger, 'Icon')) ||
+    isComponentGlyph(trigger)
+  ) {
+    console.warn(
+      'Using a LeafyGreenUI Icon or Glyph component as a trigger will not render a Tooltip, as these components do not render their children. To use, please wrap your trigger element in another HTML tag.',
+    );
+  }
```

Comment:
could be good to have some unit tests. Select.spec.tsx has some examples

Reply:
Yep, I like that idea. Updating now.